paddle/fluid/operators/mkldnn/fc_mkldnn_op.cc

```diff
@@ -410,7 +416,8 @@ class FCMKLDNNKernel : public framework::OpKernel<T_in> {
         phi::funcs::CreateKey(dev_ctx,
                               ctx.InputName("Input"),
                               ctx.InputName("W"),
-                              phi::vectorize(x->dims())));
+                              phi::vectorize(x->dims()),
+                              phi::vectorize(weights->dims())));
```

Comment:
do we need to also encode src/weight layout information to this cache key? how about adding a test case for same shape but different layout.